api/mail.go

```diff
@@ -280,3 +306,30 @@ func (a *API) validateEmail(ctx context.Context, email string) error {
 	}
 	return nil
 }
+
+// generateUniqueEmailOtp returns a unique otp
+func generateUniqueEmailOtp(tx *storage.Connection, tokenType string) (string, error) {
```

Comment:
/nit can consider changing tokenType to an enum in the future to guard against accidental typos on the `tokenType` field

Reply:
that's a good suggestion, i'll make this change in the same PR!